tests/unit/test_serialize.py

```diff
@@ -737,3 +737,71 @@ def test_invalid_timestamp_precision_raises_error(self):
                 timestamp_precision='invalid',
             )
         self.assertIn("Invalid timestamp precision", str(context.exception))
+
+
+class TestRpcV2CBORHostPrefix(unittest.TestCase):
+    def setUp(self):
+        self.model = {
+            'metadata': {
+                'protocol': 'smithy-rpc-v2-cbor',
+                'apiVersion': '2014-01-01',
+                'serviceId': 'MyService',
+                'targetPrefix': 'sampleservice',
+                'documentation': '',
+            },
+            'operations': {
+                'TestOperation': {
+                    'name': 'TestOperation',
+                    'http': {
+                        'method': 'POST',
+                        'requestUri': '/',
+                    },
+                    'input': {'shape': 'InputShape'},
+                }
+            },
+            'shapes': {
+                'InputShape': {
+                    'type': 'structure',
+                    'members': {
+                        'Foo': {'shape': 'StringType'},
+                    },
+                },
+                'StringType': {
+                    'type': 'string',
+                },
+            },
+        }
+        self.service_model = ServiceModel(self.model)
+
+    def test_host_prefix_added_to_serialized_request(self):
+        self.model['operations']['TestOperation']['endpoint'] = {
```

Comment:
Instead of adding to the operation dynamically for this test, can we just make two operations like `TestNoHostPrefixOperation` and `TestHostPrefixOperation`?

If we do this, would also consolidate this into a single test with pymark.parametrize, but that's up to you.  Keeping them separate is fine too.